src/nvim/lua/vim.lua

```diff
@@ -120,10 +120,21 @@ function vim._load_package(name)
   end
 
   for _,trail in ipairs(vim._so_trails) do
-    local path = "lua/"..trail:gsub('?',basename)
+    local path = "lua"..trail:gsub('?',basename) -- so_trails contains a leading slash
     local found = vim.api.nvim_get_runtime_file(path, false)
     if #found > 0 then
-      return package.loadlib(found[1])
+      -- Get the module name from a path. require('subdir/mymod') -> 'mymod'
```

Comment:
the require syntax is `require'subdir.mymod'` we translate it to `/` ourselves (just like the builtin finder does)